### ISSUE TITLE
Imported types (vue 3.3 feature) not working

### ISSUE THREAD

**Issue by Rigo-m** [nuxt/nuxt#20936]
### Environment

https://stackblitz.com/edit/github-ybuvkt?file=components/Test.vue

### Reproduction

https://stackblitz.com/edit/github-ybuvkt?file=components/Test.vue

### Describe the bug

Vite-node crashes when loading a component that imports types from another file (vue 3.3 feature)

### Additional context

Stackblitz gives an error related to the import of a typescript file, but the issue is easily reproducible locally with the same 3 files (Test.vue, test.props.ts and adding Test to the App.vue file) after nuxi init a nuxt 3.5 project

**Comment by Rigo-m** [comment]
Console says: Cannot read property of undefined (reading 'sys')

**Comment by Rigo-m** [comment]
> Using a non-imported interface, you code works
> ```
> <template>
>   {{ props.title }}
> </template>
> 
> <script setup lang="ts">
> interface TestProps {
>   title?: string;
> }
> 
> const props = defineProps<TestProps>();
> </script>
> ```

That's the whole point of the issue, imported types are a new feature of Vue 3.3, which have been implemented in Nuxt 3.5.

**Comment by Rigo-m** [comment]
@danielroe is this fixable within nuxt or is it an upstream issue with vite-node?

**Comment by danielroe** [comment]
It may need to be implemented in `unimport` but IIRC we have implemented the necessary utility in `mlly`.

**Comment by Rigo-m** [comment]
Let me know if I can be of any further help. It's quite unfortunate that it's not working for now

**Comment by danielroe** [comment]
Ah, misread the issue in my last reply. (Was referring to auto import of types.)

Yes, I agree.

**Comment by yassilah** [comment]
@Rigo-m So your reproduction in Stackblitz works fine by just setting the compiler to use `node:fs` to resolve/read the files

https://stackblitz.com/edit/github-ybuvkt-sqzx3w?file=nuxt.config.ts

**Comment by Rigo-m** [comment]
What's the default resolution engine for juxt vite configuration? 🤔
Wondering if this can have side effects

**Comment by Rigo-m** [comment]
> @Rigo-m So your reproduction in Stackblitz works fine by just setting the compiler to use `node:fs` to resolve/read the files
> 
> https://stackblitz.com/edit/github-ybuvkt-sqzx3w?file=nuxt.config.ts

This works in a big repo we have, so anecdotically speaking it seems to be pretty safe as an option for now.

**Comment by mrc-bsllt** [comment]
> @Rigo-m So your reproduction in Stackblitz works fine by just setting the compiler to use `node:fs` to resolve/read the files
> 
> https://stackblitz.com/edit/github-ybuvkt-sqzx3w?file=nuxt.config.ts

I have the same issue with a Nuxt 3.5.1 project, but this solution doesn't work for me. I have the following error: [plugin:vite:vue] EISDIR: illegal operation on a directory, read

**Comment by Rigo-m** [comment]
> I have the same issue with a Nuxt 3.5.1 project, but this solution doesn't work for me. I have the following error: [plugin:vite:vue] EISDIR: illegal operation on a directory, read

You need to raise your OS simultaneously opened file limits, ask google or chatGPT on how to do it

**Comment by mrc-bsllt** [comment]
> > I have the same issue with a Nuxt 3.5.1 project, but this solution doesn't work for me. I have the following error: [plugin:vite:vue] EISDIR: illegal operation on a directory, read
> 
> You need to raise your OS simultaneously opened file limits, ask google or chatGPT on how to do it

No, in my case the problem was due to my error in importing the .ts file.

before:
`import type { TestNewProps } from './types';`

after:
`import type { TestNewProps } from './types/index';`

**Comment by danielroe** [comment]
Try installing `typescript` as a project dependency.

**Comment by acicero93** [comment]
> Try installing `typescript` as a project dependency.

This solved it for me. Thanks!

**Comment by danielroe** [comment]
We should figure out why there is no useful log being displayed from Vue, and either file an issue upstream or add it to the documentation (or both).

**Comment by sxzz** [comment]
I will add a warning in Vue when typescript is not installed. But maybe we can do something on Nuxt side: add `typescript` as a dev dep into the nuxt starter template.

**Comment by Rigo-m** [comment]
Thanks @sxzz , going to close this issue as adding typescript as a dependency worked.

**Comment by awacode21** [comment]
i now also run into this issue. Adding typescript as a dependency works. But like me i guess every single user that wants to use that feature in nuxt will run into the problem and will need to research for the bug. Where there any docs enhanced? Or could it be solved by nuxt framework adding the dependency itself?

**Comment by danielroe** [comment]
I think it would be very reasonable. We don't want to ship TS as a dependency given the additional weight, but if there's a way we could know that the user needs it an offer to auto-install it, that would be perfect.

**Comment by mathiasayivor** [comment]
I ran into this issue earlier today and my findings led me here. I think it would be worth it to add it to the installation docs.

**Comment by muneebbug** [comment]
> Try installing `typescript` as a project dependency.

This solved the issue for me 🤞

**Comment by devinschumacher** [comment]
> ### Environment
> https://stackblitz.com/edit/github-ybuvkt?file=components/Test.vue
> 
> ### Reproduction
> https://stackblitz.com/edit/github-ybuvkt?file=components/Test.vue
> 
> ### Describe the bug
> Vite-node crashes when loading a component that imports types from another file (vue 3.3 feature)
> 
> ### Additional context
> Stackblitz gives an error related to the import of a typescript file, but the issue is easily reproducible locally with the same 3 files (Test.vue, test.props.ts and adding Test to the App.vue file) after nuxi init a nuxt 3.5 project
> 
> ### Logs
> _No response_

It work's if you use `pnpm` instead of `npm`!

**Comment by ponochovny** [comment]
`"typescript": "^7.0.2"` breaks my `shadcn-vue` components... but with `"typescript": "~5.8.2"` it works fine

**Comment by danielroe** [comment]
typescript 7 does not yet have support for vue
